App.js

```diff
@@ -6,38 +6,488 @@ import styles from './src/style';
-
-	async componentDidMount() {
-		await Font.loadAsync({
-			'neutraface': require('./assets/fonts/NeutraText-Bold.otf'),
-			'open-sans': require('./assets/fonts/OpenSans-Regular.ttf')
-		});
-		this.setState({loaded: true});
-	}
-
-	render() {
-		if (!this.state.loaded) {
-			return (
-				<View>
-					<ActivityIndicator/>
-				</View>
-			);
-		}
-
-
-		return (
-			<View style={styles.container}>
-				<RouteSelection/>
-			</View>
-		);
-	}
+export default class BuildingMarker extends Component {
+  
+  constructor(props){
+          super(props);
+          this.state = {isLoading: true, scrollable: false , data:[]};
+     }
+  
+     componentDidMount(){
+         var paths = this.props.filenames.reduce(
+  ["10Church",
+     "amundson",
+     "universityAveRamp",
+     "eastRiverGarage",
+     "ford",
+     "admundson",
+     "akerman",
+     "diehl",
+     "boynton",
+     "dVC",
+     "civilE",
+     "childrenRehabClinic",
+     "hsec",
+     "keller",
+     "lind",
+     "mayoBuilding",
+     "mcb",
+     "moosTower",
+     "nilsHasselmo",
+     "nolteCtr",
+     "northrop",
+     "phillipsWangensteen",
+     "rapson",
+     "shepherdLab",
+     "tate",
+     "vincent",
+     "hospitalRamp",
+     "washingtonAveRamp",
+     "weaverDensford",
+     "akerman",
+     "coffman",
+     "folwell",
+     "graduate",
+     "hasselmo",
+     "jackson",
+     "jones",
+     "keller",
+     "kolthoff",
+     "lind",
+     "mCC",
+     "mcNamara",
+     "mechE",
+     "smith",
+     "nolteCtrGarage",
+     "10Church",
+     "folwell",
+     "morill",
+     "murphy",
+     "nolteCtr",
+     "northropGarage",
+     "tate",
+     "tNs",
+     "universityAveRamp",
+     "varClub",
+     "walter",
+     "washingtonAveRamp",
+     "williamson",
+     "yudof",
+     "johnston",
+     "mayoBldg"]);
+          var urls = paths.map(path => "https://sase-labs-2020.github.io/assets/informations/" + path.join("_") + ".json");
+          urls = urls.forEach(url =>
+               {return fetch(url,
+                         {
+                              method: "GET",
+                              headers: {
+                              'Accept': 'application/json',
+                              'Content-Type': 'application/json',
+                              },
+                         }
+                    ).then(response => response.json())
+                    .then((responseData) => {
+                         console.log(responseData);
+                              this.setState(
+                                   (prevState) => {
+                                        return {
+                                             data: prevState.data.concat(responseData),
+                                             isLoading: false,
+                                        };
+                                   }
+                              );
+                         });
+          });
+     }
+  
+  
+render() {
+  return (
+    <MapView
```

Comment:
As mentioned in the meeting, please don't hard-code each possible component individually. Please see [this line](https://github.com/SASE-Labs-2020/gopherway/blob/master/src/components/GraphEdge.js#L94) to see how to leverage higher-order functions for that. Keep in mind that this pattern was enabled by [a step in the `componentDidMount`](https://github.com/SASE-Labs-2020/gopherway/blob/master/src/components/GraphEdge.js#L72).